File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-confirmation-processes-5-4/RequestCollateralAllocation.cpp
// RequestCollateralAllocation.cpp 
#include "RequestCollateralAllocation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

RequestCollateralAllocation::RequestCollateralAllocation(TiXmlNode* xmlNode)
: CorrectableRequestMessage(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //serviceNameNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* serviceNameNode = xmlNode->FirstChildElement("serviceName");

   if(serviceNameNode){serviceNameIsNull_ = false;}
   else{serviceNameIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- serviceNameNode , address : " << serviceNameNode << std::endl;
   #endif
   if(serviceNameNode)
   {
      if(serviceNameNode->Attribute("href") || serviceNameNode->Attribute("id"))
      {
          if(serviceNameNode->Attribute("id"))
          {
             serviceNameIDRef_ = serviceNameNode->Attribute("id");
             serviceName_ = boost::shared_ptr<XsdTypeNormalizedString>(new XsdTypeNormalizedString(serviceNameNode));
             serviceName_->setID(serviceNameIDRef_);
             IDManager::instance().setID(serviceNameIDRef_,serviceName_);
          }
          else if(serviceNameNode->Attribute("href")) { serviceNameIDRef_ = serviceNameNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { serviceName_ = boost::shared_ptr<XsdTypeNormalizedString>(new XsdTypeNormalizedString(serviceNameNode));}
   }

   //collateralGiverPartyReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* collateralGiverPartyReferenceNode = xmlNode->FirstChildElement("collateralGiverPartyReference");

   if(collateralGiverPartyReferenceNode){collateralGiverPartyReferenceIsNull_ = false;}
   else{collateralGiverPartyReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- collateralGiverPartyReferenceNode , address : " << collateralGiverPartyReferenceNode << std::endl;
   #endif
   if(collateralGiverPartyReferenceNode)
   {
      if(collateralGiverPartyReferenceNode->Attribute("href") || collateralGiverPartyReferenceNode->Attribute("id"))
      {
          if(collateralGiverPartyReferenceNode->Attribute("id"))
          {
             collateralGiverPartyReferenceIDRef_ = collateralGiverPartyReferenceNode->Attribute("id");
             collateralGiverPartyReference_ = boost::shared_ptr<PartyReference>(new PartyReference(collateralGiverPartyReferenceNode));
             collateralGiverPartyReference_->setID(collateralGiverPartyReferenceIDRef_);
             IDManager::instance().setID(collateralGiverPartyReferenceIDRef_,collateralGiverPartyReference_);
          }
          else if(collateralGiverPartyReferenceNode->Attribute("href")) { collateralGiverPartyReferenceIDRef_ = collateralGiverPartyReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { collateralGiverPartyReference_ = boost::shared_ptr<PartyReference>(new PartyReference(collateralGiverPartyReferenceNode));}
   }

   //collateralAllocationNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* collateralAllocationNode = xmlNode->FirstChildElement("collateralAllocation");

   if(collateralAllocationNode){collateralAllocationIsNull_ = false;}
   else{collateralAllocationIsNull_ = true;}

   if(collateralAllocationNode)
   {
      for(collateralAllocationNode; collateralAllocationNode; collateralAllocationNode = collateralAllocationNode->NextSiblingElement("collateralAllocation")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- collateralAllocationNode , address : " << collateralAllocationNode << std::endl;
          #endif
          if(collateralAllocationNode->Attribute("href") || collateralAllocationNode->Attribute("id"))
          {
              if(collateralAllocationNode->Attribute("id"))
              {
                  collateralAllocationIDRef_ = collateralAllocationNode->Attribute("id");
                  collateralAllocation_.push_back(boost::shared_ptr<ProposedCollateralAllocation>(new ProposedCollateralAllocation(collateralAllocationNode)));
                  collateralAllocation_.back()->setID(collateralAllocationIDRef_);
                  IDManager::instance().setID(collateralAllocationIDRef_, collateralAllocation_.back());
              }
              else if(collateralAllocationNode->Attribute("href")) { collateralAllocationIDRef_ = collateralAllocationNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { collateralAllocation_.push_back(boost::shared_ptr<ProposedCollateralAllocation>(new ProposedCollateralAllocation(collateralAllocationNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- collateralAllocationNode , address : " << collateralAllocationNode << std::endl;
       #endif
   }

   //partyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* partyNode = xmlNode->FirstChildElement("party");

   if(partyNode){partyIsNull_ = false;}
   else{partyIsNull_ = true;}

   if(partyNode)
   {
      for(partyNode; partyNode; partyNode = partyNode->NextSiblingElement("party")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partyNode , address : " << partyNode << std::endl;
          #endif
          if(partyNode->Attribute("href") || partyNode->Attribute("id"))
          {
              if(partyNode->Attribute("id"))
              {
                  partyIDRef_ = partyNode->Attribute("id");
                  party_.push_back(boost::shared_ptr<Party>(new Party(partyNode)));
                  party_.back()->setID(partyIDRef_);
                  IDManager::instance().setID(partyIDRef_, party_.back());
              }
              else if(partyNode->Attribute("href")) { partyIDRef_ = partyNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { party_.push_back(boost::shared_ptr<Party>(new Party(partyNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partyNode , address : " << partyNode << std::endl;
       #endif
   }

   //accountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* accountNode = xmlNode->FirstChildElement("account");

   if(accountNode){accountIsNull_ = false;}
   else{accountIsNull_ = true;}

   if(accountNode)
   {
      for(accountNode; accountNode; accountNode = accountNode->NextSiblingElement("account")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- accountNode , address : " << accountNode << std::endl;
          #endif
          if(accountNode->Attribute("href") || accountNode->Attribute("id"))
          {
              if(accountNode->Attribute("id"))
              {
                  accountIDRef_ = accountNode->Attribute("id");
                  account_.push_back(boost::shared_ptr<Account>(new Account(accountNode)));
                  account_.back()->setID(accountIDRef_);
                  IDManager::instance().setID(accountIDRef_, account_.back());
              }
              else if(accountNode->Attribute("href")) { accountIDRef_ = accountNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { account_.push_back(boost::shared_ptr<Account>(new Account(accountNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- accountNode , address : " << accountNode << std::endl;
       #endif
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeNormalizedString> RequestCollateralAllocation::getServiceName()
{
   if(!this->serviceNameIsNull_){
        if(serviceNameIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeNormalizedString>(IDManager::instance().getID(serviceNameIDRef_));
        }else{
             return this->serviceName_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeNormalizedString>();
   }
}
boost::shared_ptr<PartyReference> RequestCollateralAllocation::getCollateralGiverPartyReference()
{
   if(!this->collateralGiverPartyReferenceIsNull_){
        if(collateralGiverPartyReferenceIDRef_ != ""){
             return boost::shared_static_cast<PartyReference>(IDManager::instance().getID(collateralGiverPartyReferenceIDRef_));
        }else{
             return this->collateralGiverPartyReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PartyReference>();
   }
}
std::vector<boost::shared_ptr<ProposedCollateralAllocation>> RequestCollateralAllocation::getCollateralAllocation()
{
   if(!this->collateralAllocationIsNull_){
        if(collateralAllocationIDRef_ != ""){
             return this->collateralAllocation_;
        }else{
             return this->collateralAllocation_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<ProposedCollateralAllocation>>();
   }
}
std::vector<boost::shared_ptr<Party>> RequestCollateralAllocation::getParty()
{
   if(!this->partyIsNull_){
        if(partyIDRef_ != ""){
             return this->party_;
        }else{
             return this->party_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<Party>>();
   }
}
std::vector<boost::shared_ptr<Account>> RequestCollateralAllocation::getAccount()
{
   if(!this->accountIsNull_){
        if(accountIDRef_ != ""){
             return this->account_;
        }else{
             return this->account_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<Account>>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-confirmation-processes-5-4/RequestCollateralAllocation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class RequestCollateralAllocation : CorrectableRequestMessage
    {
        public RequestCollateralAllocation(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode serviceNameNode = xmlNode.SelectSingleNode("serviceName");
            
            if (serviceNameNode != null)
            {
                if (serviceNameNode.Attributes["href"] != null || serviceNameNode.Attributes["id"] != null) 
                {
                    if (serviceNameNode.Attributes["id"] != null) 
                    {
                        serviceNameIDRef_ = serviceNameNode.Attributes["id"].Value;
                        XsdTypeNormalizedString ob = new XsdTypeNormalizedString(serviceNameNode);
                        IDManager.SetID(serviceNameIDRef_, ob);
                    }
                    else if (serviceNameNode.Attributes["href"] != null)
                    {
                        serviceNameIDRef_ = serviceNameNode.Attributes["href"].Value;
                    }
                    else
                    {
                        serviceName_ = new XsdTypeNormalizedString(serviceNameNode);
                    }
                }
                else
                {
                    serviceName_ = new XsdTypeNormalizedString(serviceNameNode);
                }
            }
            
        
            XmlNode collateralGiverPartyReferenceNode = xmlNode.SelectSingleNode("collateralGiverPartyReference");
            
            if (collateralGiverPartyReferenceNode != null)
            {
                if (collateralGiverPartyReferenceNode.Attributes["href"] != null || collateralGiverPartyReferenceNode.Attributes["id"] != null) 
                {
                    if (collateralGiverPartyReferenceNode.Attributes["id"] != null) 
                    {
                        collateralGiverPartyReferenceIDRef_ = collateralGiverPartyReferenceNode.Attributes["id"].Value;
                        PartyReference ob = new PartyReference(collateralGiverPartyReferenceNode);
                        IDManager.SetID(collateralGiverPartyReferenceIDRef_, ob);
                    }
                    else if (collateralGiverPartyReferenceNode.Attributes["href"] != null)
                    {
                        collateralGiverPartyReferenceIDRef_ = collateralGiverPartyReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        collateralGiverPartyReference_ = new PartyReference(collateralGiverPartyReferenceNode);
                    }
                }
                else
                {
                    collateralGiverPartyReference_ = new PartyReference(collateralGiverPartyReferenceNode);
                }
            }
            
        
            XmlNodeList collateralAllocationNodeList = xmlNode.SelectNodes("collateralAllocation");
            
            if (collateralAllocationNodeList != null)
            {
                this.collateralAllocation_ = new List<ProposedCollateralAllocation>();
                foreach (XmlNode item in collateralAllocationNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            collateralAllocationIDRef_ = item.Attributes["id"].Value;
                            collateralAllocation_.Add(new ProposedCollateralAllocation(item));
                            IDManager.SetID(collateralAllocationIDRef_, collateralAllocation_[collateralAllocation_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            collateralAllocationIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        collateralAllocation_.Add(new ProposedCollateralAllocation(item));
                        }
                    }
                    else
                    {
                        collateralAllocation_.Add(new ProposedCollateralAllocation(item));
                    }
                }
            }
            
        
            XmlNodeList partyNodeList = xmlNode.SelectNodes("party");
            
            if (partyNodeList != null)
            {
                this.party_ = new List<Party>();
                foreach (XmlNode item in partyNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            partyIDRef_ = item.Attributes["id"].Value;
                            party_.Add(new Party(item));
                            IDManager.SetID(partyIDRef_, party_[party_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            partyIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        party_.Add(new Party(item));
                        }
                    }
                    else
                    {
                        party_.Add(new Party(item));
                    }
                }
            }
            
        
            XmlNodeList accountNodeList = xmlNode.SelectNodes("account");
            
            if (accountNodeList != null)
            {
                this.account_ = new List<Account>();
                foreach (XmlNode item in accountNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            accountIDRef_ = item.Attributes["id"].Value;
                            account_.Add(new Account(item));
                            IDManager.SetID(accountIDRef_, account_[account_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            accountIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        account_.Add(new Account(item));
                        }
                    }
                    else
                    {
                        account_.Add(new Account(item));
                    }
                }
            }
            
        
        }
        
    
        #region ServiceName_
        private XsdTypeNormalizedString serviceName_;
        public XsdTypeNormalizedString ServiceName_
        {
            get
            {
                if (this.serviceName_ != null)
                {
                    return this.serviceName_; 
                }
                else if (this.serviceNameIDRef_ != null)
                {
                    serviceName_ = IDManager.getID(serviceNameIDRef_) as XsdTypeNormalizedString;
                    return this.serviceName_; 
                }
                else
                {
                      return this.serviceName_; 
                }
            }
            set
            {
                if (this.serviceName_ != value)
                {
                    this.serviceName_ = value;
                }
            }
        }
        #endregion
        
        public string serviceNameIDRef_ { get; set; }
        #region CollateralGiverPartyReference_
        private PartyReference collateralGiverPartyReference_;
        public PartyReference CollateralGiverPartyReference_
        {
            get
            {
                if (this.collateralGiverPartyReference_ != null)
                {
                    return this.collateralGiverPartyReference_; 
                }
                else if (this.collateralGiverPartyReferenceIDRef_ != null)
                {
                    collateralGiverPartyReference_ = IDManager.getID(collateralGiverPartyReferenceIDRef_) as PartyReference;
                    return this.collateralGiverPartyReference_; 
                }
                else
                {
                      return this.collateralGiverPartyReference_; 
                }
            }
            set
            {
                if (this.collateralGiverPartyReference_ != value)
                {
                    this.collateralGiverPartyReference_ = value;
                }
            }
        }
        #endregion
        
        public string collateralGiverPartyReferenceIDRef_ { get; set; }
        #region CollateralAllocation_
        private List<ProposedCollateralAllocation> collateralAllocation_;
        public List<ProposedCollateralAllocation> CollateralAllocation_
        {
            get
            {
                if (this.collateralAllocation_ != null)
                {
                    return this.collateralAllocation_; 
                }
                else if (this.collateralAllocationIDRef_ != null)
                {
                    return this.collateralAllocation_; 
                }
                else
                {
                      return this.collateralAllocation_; 
                }
            }
            set
            {
                if (this.collateralAllocation_ != value)
                {
                    this.collateralAllocation_ = value;
                }
            }
        }
        #endregion
        
        public string collateralAllocationIDRef_ { get; set; }
        #region Party_
        private List<Party> party_;
        public List<Party> Party_
        {
            get
            {
                if (this.party_ != null)
                {
                    return this.party_; 
                }
                else if (this.partyIDRef_ != null)
                {
                    return this.party_; 
                }
                else
                {
                      return this.party_; 
                }
            }
            set
            {
                if (this.party_ != value)
                {
                    this.party_ = value;
                }
            }
        }
        #endregion
        
        public string partyIDRef_ { get; set; }
        #region Account_
        private List<Account> account_;
        public List<Account> Account_
        {
            get
            {
                if (this.account_ != null)
                {
                    return this.account_; 
                }
                else if (this.accountIDRef_ != null)
                {
                    return this.account_; 
                }
                else
                {
                      return this.account_; 
                }
            }
            set
            {
                if (this.account_ != value)
                {
                    this.account_ = value;
                }
            }
        }
        #endregion
        
        public string accountIDRef_ { get; set; }
        
    
        
    
    }
    
}

